cunumeric/module.py

```diff
@@ -6227,3 +6227,719 @@ def bincount(
+    k = floor(pos)
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# Johnson & Kotz method:
+# 'median_unbiased'
+#
+def median_unbiased(q, n):
+    fract = 1.0 / 3.0
+    pos = q * (n + fract) + fract
+    k = floor(pos)
+
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# Blom method:
+# 'normal_unbiased'
+#
+def normal_unbiased(q, n):
+    fract1 = 0.25
+    fract2 = 3.0 / 8.0
+    pos = q * (n + fract1) + fract2
+    k = floor(pos)
+
+    # gamma = pos-k
+    #
+    # this fixes it:
+    # (when pos > n: this actually selects the right point,
+    #  which is the correct choice, because right = arr[n]
+    #  gets invalidated)
+    #
+    gamma = 0.0 if (pos < 1 or pos > n) else pos - k
+
+    j = floor_i(k)
+    return (gamma, j)
+
+
+# `lower`
+#
+def lower(q, n):
+    gamma = 0.0
+    pos = q * (n - 1)
+    k = floor(pos)
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `higher`
+#
+def higher(q, n):
+    pos = q * (n - 1)
+    k = floor(pos)
+
+    # Generic rule: (k == pos)
+    gamma = 0.0 if (pos == 0 or k == pos) else 1.0
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `midpoint`
+#
+def midpoint(q, n):
+    pos = q * (n - 1)
+    k = floor(pos)
+
+    # Generic rule: (k == pos)
+    gamma = 0.0 if (pos == 0 or k == pos) else 0.5
+
+    j = int(k)
+    return (gamma, j)
+
+
+# `nearest`
+#
+def nearest(q, n):
+    pos = q * (n - 1)
+
+    # k = floor(pos)
+    # gamma = 1.0 if pos - k >= 0.5 else 0.0
+
+    k = np.round(pos)
+    gamma = 0.0
+
+    j = int(k)
+    return (gamma, j)
+
+
+# helper: test array if sorted:
+#
+def is_sorted(a):
+    return all(a[:-1] <= a[1:])
+
+
+# helper: test if array inputs are different:
+#
+def is_diff(arr1, arr2):
+    return len(arr1) != len(arr2) or any(
+        [arr1[i] != arr2[i] for i in range(0, len(arr1))]
+    )
+
+
+# for the case when axis = tuple (non-singleton)
+# reshuffling might have to be done (if tuple is non-consecutive)
+# and the src array must be collapsed along that set of axes
+#
+# args:
+#
+# arr:    [in] source nd-array on which quantiles are calculated;
+# axes_set: [in] tuple or list of axes (indices less than arr dimension);
+#
+# return: pair: (minimal_index, reshuffled_and_collapsed source array)
+# TODO: check if reshuffling, reshaping is done in-place!
+#
+def reshuffle_reshape(arr, axes_set):
+    ndim = len(arr.shape)
+
+    if not is_sorted(axes_set):
+        sorted_axes = tuple(sort(axes_set))
+    else:
+        sorted_axes = tuple(axes_set)
```

Comment:
I think just `sorted_axes = tuple(sorted(axes_set))` is fine here.

Tyipcally `axes_set` will be a tuple of <5 elements; doing the "is sorted" check is likely to take a similar amount of time to just sorting it.

Especially going through `cunumeric.sort`, as the current code is doing, which has the potential to invoke all the distributed machinery of Legion, is overkill here.

`sorted` will just use the built-in python sorting, that works on lists, tuples etc.

Reply:
Done.